src/pages/resultsView/enrichments/AlterationEnrichmentsContainer.tsx

```diff
@@ -145,49 +124,64 @@ export default class AlterationEnrichmentContainer extends React.Component<IAlte
         }
     );
 
-    @computed get columns() {
+    @computed get customColumns() {
+        return getGroupColumns(this.props.groups, this.props.alteredVsUnalteredMode);
+    }
+
+    @computed get visibleOrderedColumnNames() {
         const columns = [];
         columns.push(AlterationEnrichmentTableColumnType.GENE,
             AlterationEnrichmentTableColumnType.CYTOBAND);
         if (this.props.showCNAInTable) {
             columns.push(AlterationEnrichmentTableColumnType.ALTERATION);
         }
-        columns.push(AlterationEnrichmentTableColumnType.PERCENTAGE_IN_GROUP1,
-        AlterationEnrichmentTableColumnType.PERCENTAGE_IN_GROUP2,
-        AlterationEnrichmentTableColumnType.LOG_RATIO,
+        this.props.groups.forEach(group=>{
+            columns.push(group.name);
+        })
+        if(this.isTwoGroupAnalysesPossible) {
+            columns.push("Log Ratio");
+        }
+        
+        columns.push(
         AlterationEnrichmentTableColumnType.P_VALUE,
-        AlterationEnrichmentTableColumnType.Q_VALUE,
-        AlterationEnrichmentTableColumnType.TENDENCY);
+        AlterationEnrichmentTableColumnType.Q_VALUE);
+
+        if(this.isTwoGroupAnalysesPossible) {
+            columns.push(this.props.alteredVsUnalteredMode ? "Tendency" : "Enriched in");
+        }
 
         return columns;
     }
 
     @computed get volcanoPlotLabels() {
-        if (this.props.alteredVsUnalteredMode) {
-            return ["Mutual exclusivity", "Co-occurrence"];
-        } else {
-            return [this.props.group2Name!, this.props.group1Name!];
+        if(this.props.groups.length === 2) {
+            let lable1 = this.props.groups[0].displayName ? this.props.groups[0].displayName : this.props.groups[0].name;
+            let lable2 = this.props.groups[1].displayName ? this.props.groups[1].displayName : this.props.groups[1].name;
```

Comment:
btw this can be written `this.props.group[1].displayName || this.props.groups[1].name`